File: selectionSort/selectionSort.java
import java.util.Random;

class SelectionSort
{
    public static void main(String[] args)
    {
	int arraySize = 100000;
	int[] toSort = new int[arraySize];
	Random rand = new Random(arraySize);
	for(int j = 0; j < 20; j++)
	    {
		for(int i = 0; i < toSort.length; i++)
		    {
			toSort[i] = rand.nextInt(arraySize * 10);
		    }
		Long start = System.nanoTime();
		sort(toSort);
		Long end = System.nanoTime();
		System.out.println("Sorting took: " + (end - start));

	    }
    }

    private static int[] sort(int[] toSort)
    {
	//Instantiant the two required ints
	int smallestEntry;
	int smallestEntryIndex;
	//Loop over all elements (last one not needed)
	for(int i = 0; i < toSort.length - 1; i++)
	    {
		smallestEntry = toSort[i];
		smallestEntryIndex = i;

		//See if we need to switch
		for(int j = i + 1; j < toSort.length; j++)
		    {
			if(smallestEntry > toSort[j])
			    {
				smallestEntry = toSort[j];
				smallestEntryIndex = j;
			    }
		    }
		toSort[smallestEntryIndex] = toSort[i];
		toSort[i] = smallestEntry;	    
	    }
	return toSort;
    }
}

File: selectionSort/selectionSort.c
#include <stdlib.h>
#include <stdio.h>
#include <time.h>

void selectionSort(int *toSort, int size);
static unsigned long getNanos(void);

int main(void)
{
  time_t t1,t2;
  int elapsedTime;
  int arraySize = 100000;
  int toSort[arraySize];
  srand((unsigned int)arraySize);
  unsigned long start,end;

  for(int i = 0; i < 20; i++)
    {
      for(int j = 0; j < arraySize; j++)
	{
	  toSort[j] = rand() % (arraySize * 10);
	}

      start = getNanos();
      selectionSort(toSort, arraySize);
      end = getNanos();

      fprintf(stdout, "Sorting took %ld\n", (end-start));

    }
  return 1;
}

void selectionSort(int *toSort, int size)
{
  int entryIndex;
  int entryVal;
  
  //Loop over the array
  for(int i =0; i < size - 1; i++)
    {
      entryIndex = i;
      entryVal = toSort[i];
      
      //Loop over the remaining items 
      for(int j = i; j < size; j++)
	{
	  if(toSort[j] < entryVal)
	    {
	      entryIndex = j;
	      entryVal = toSort[j];
	    }
	}
      toSort[entryIndex] = toSort[i];
      toSort[i] = entryVal;
    }
}

static unsigned long getNanos(void) {
  struct timespec ts;
  timespec_get(&ts, TIME_UTC);
  return (unsigned long)ts.tv_sec * 1000000000L + ts.tv_nsec;
}
